File: java/com/lc/lc19.java
package lc;

public class lc19 {
    public ListNode removeNthFromEnd(ListNode head, int n) {
        int k = n;
        ListNode dummy = new ListNode(-1, head);
        ListNode fast = dummy, slow = dummy;
        while (k > 0) {
            fast = fast.next;
            k--;
        }
        while (fast.next != null) {
            fast = fast.next;
            slow = slow.next;
        }
        slow.next = slow.next.next;
        return dummy.next;
    }
}

File: go/lc/lc19.go
package main

func removeNthFromEnd(head *ListNode, n int) *ListNode {
	dummy := &ListNode{Next: head}
	fi, se := dummy, dummy
	for i := 0; i < n; i++ {
		fi = fi.Next
	}
	for fi.Next != nil {
		fi = fi.Next
		se = se.Next
	}
	se.Next = se.Next.Next
	return dummy.Next
}
